daemon/algod/api/server/v2/utils.go

```diff
@@ -360,11 +360,59 @@ func ConvertInnerTxn(txn *transactions.SignedTxnWithAD) PreEncodedTxInfo {
 	return response
 }
 
+func convertTxnTrace(txnTrace *simulation.TransactionTrace) *model.SimulationTransactionExecTrace {
+	if txnTrace == nil {
+		return nil
+	}
+
+	var innerTraces []model.SimulationTransactionExecTrace
+	for i := range txnTrace.InnerTraces {
+		innerTrace := *convertTxnTrace(&txnTrace.InnerTraces[i])
+		innerTraces = append(innerTraces, innerTrace)
+	}
+	var innerTracesPtr *[]model.SimulationTransactionExecTrace
+	if len(innerTraces) > 0 {
+		innerTracesPtr = &innerTraces
+	}
+
+	var opcodeTrace []model.SimulationOpcodeTraceUnit
+	for i := range txnTrace.Trace {
+		opcodeTrace = append(opcodeTrace, model.SimulationOpcodeTraceUnit{Pc: txnTrace.Trace[i].PC})
+	}
+
+	var pcToInner []model.SimulationPcToInnerIndex
+	for k, v := range txnTrace.StepToInnerMap {
+		pcToInner = append(pcToInner, model.SimulationPcToInnerIndex{Pc: k, InnerIndex: uint64(v)})
+	}
+	var pcToInnerPtr *[]model.SimulationPcToInnerIndex
+	if len(pcToInner) > 0 {
+		pcToInnerPtr = &pcToInner
+	}
+
+	var traceType model.SimulationTransactionExecTraceTraceType
+	switch txnTrace.TraceType {
+	case simulation.AppCallApprovalTransaction:
+		traceType = model.SimulationTransactionExecTraceTraceTypeApprovalProgram
+	case simulation.AppCallClearStateTransaction:
+		traceType = model.SimulationTransactionExecTraceTraceTypeClearStateProgram
+	case simulation.LogicSigTransaction:
+		traceType = model.SimulationTransactionExecTraceTraceTypeLogicSignature
+	}
+
+	return &model.SimulationTransactionExecTrace{
+		TraceType:      traceType,
+		Trace:          opcodeTrace,
+		InnerTrace:     innerTracesPtr,
+		StepToInnerMap: pcToInnerPtr,
+	}
```

Comment:
super nit:  order these parameter preparations in the same order as they appear in the constructor

Reply:
see [minor refactoring convertTxnTrace](https://github.com/algorand/go-algorand/pull/5360/commits/103343af101d06a42c5d0ab082a25274ed24db27)